tests/test_cloud_detector.py

```diff
@@ -5,31 +5,50 @@
 
 import numpy as np
 import pytest
+from numpy.testing import assert_allclose, assert_array_equal
 
 from s2cloudless import S2PixelCloudDetector
 
 
-def test_pixel_cloud_detector():
-    data_path = os.path.join(os.path.dirname(__file__), "TestInputs", "input_arrays.npz")
-    data = np.load(data_path)
+@pytest.fixture(name="data", scope="module")
+def data_fixture():
+    return np.load(os.path.join(os.path.dirname(__file__), "TestInputs", "input_arrays.npz"))
+
+
+@pytest.fixture(name="cloud_detector", scope="module")
+def cloud_detector_fixture():
+    return S2PixelCloudDetector(all_bands=True)
 
-    cloud_detector = S2PixelCloudDetector(all_bands=True)
-    cloud_probs = cloud_detector.get_cloud_probability_maps(data["s2_im"])
-    cloud_mask_from_prob = cloud_detector.get_mask_from_prob(cloud_probs)
-    cloud_mask = cloud_detector.get_cloud_masks(data["s2_im"])
 
-    assert np.allclose(cloud_probs, data["cl_probs"], atol=1e-14)
-    assert np.array_equal(cloud_mask, data["cl_mask"])
-    assert np.array_equal(cloud_mask_from_prob, data["cl_mask"])
+@pytest.fixture(name="cloud_detector_not_all", scope="module")
```

Comment:
updated, do not know if i like it, but looks better then before (IMO)

Reply:
at this point i'm not sure why have it as fixture at all. It would be less lines (and easier to understand) if each test had a line `cloud_detector = S2PixelCloudDetector(...)` at the start

Reply:
updated